components/sidebar_right/sidebar_right.jsx

```diff
@@ -139,12 +146,10 @@ export default class SidebarRight extends React.Component {
         }
 
         var searchForm = null;
-        if (this.props.currentUser) {
-            searchForm = <SearchBar isFocus={this.props.searchVisible}/>;
+        if (currentUser) {
+            searchForm = <SearchBar isFocus={searchVisible && !isFlaggedPosts && !isPinnedPosts}/>;
```

Comment:
Here is the actual fix.

Reply:
Can't wait for us to get the RHS state figured out properly, it's getting better but I hate seeing these kind of conditions